plugins/linux-capture/CMakeLists.txt

```diff
@@ -1,50 +1,51 @@
 project(linux-capture)
 
+option(ENABLE_PIPEWIRE "Enable PipeWire support" OFF)
+if(NOT ENABLE_PIPEWIRE)
+	message(STATUS "OBS: PipeWire support not enabled")
+endif()
+
```

Comment:
Please don't revert #4483

Reply:
Some recent changes will indeed be reverted, but for a reason:

The default configuration of the build tree should enable as few features by default as possible (some of these escaped for no, but will be forced to `OFF` by default later today) to reduce friction when configuring/building.

If you want a fully-featured configuration of OBS, a new config variable `BUILD_FOR_DISTRIBUTION` is added which enables a sane default set of features:

https://github.com/obsproject/obs-studio/blob/72d737c2f9ea5744b2903c33f037a5cf65501876/cmake/Modules/ObsDefaults_Linux.cmake#L101

This is where a feature like pipewire (which I couldn't get to work in my tests) could be enabled by default). But afaik nobody took care of adopting the CI build scripts for pipewire support either, so IMO that PR should not have been merged in the first place and which is why it's not included here.

I'll pick this up once the CI script has been updated, so that pipewire can be added to the build dependencies there and be activated by default again.

In all other cases it is upon the developer to _explicitly_ activate the features they want for their build tree using config flags or a gui like `ccmake` (shipped by default).

Reply:
Do note that I can imagine another branch inside this that can "filter" by distro if we'd like to make that "default" be different given our experience with package maintainers, I'd just need to know what you'd like as defaults there.

As it's contained in its own linux-helper script, any added complexity doesn't impact other areas.

Reply:
Some packagers don't even use git submodules to build obs-studio, nowadays only the PPA give a complete experience under linux because packager don't seem to look at the CMake options.

If they don't even look at CMake options, so the `BUILD_FOR_DISTRIBUTION` can be considered useless because they will ignore it.

They need to know they are missing something and need to make a choice, not just a "it just compile" behavior. This why @GeorgesStavracas @Gol-D-Ace welcomed/approved my PR because it force them to make a choice not just ignore it.

RPM Fusion with my changes has now integrated obs-vst (not the good way but they did it) and disabled obs-browser, before none of them were compiled
Example https://pkgs.rpmfusion.org/cgit/free/obs-studio.git/tree/obs-studio.spec

- obs-browser is now deeply link to obs-studio with things like service integration and browser docks so it needs to default to ON
- PipeWire is needed with GNOME, xdg-portal and it will replace PulseAudio in the future so make it default is not wrong
- obs-vst is a nice addition but it get always ignored.

Reply:
> They need to know they are missing something and need to make a choice, not just a "it just compile" behavior. 

I _think_ I understand the point, but to be frank this is not something that our build system should take care of, because that's _not_ what it's for and is the reason why we ended up with a convoluted web of options, some enabled, some disabled, some breaking builds if not enabled, some complaining but continuing to build, in the first place.

Usually I'd say that if people don't know what they are doing, it _should_ break. If they forget to enable features, it _should_ not work. It is not our job to do their homework.

Also from the perspective of a maintainer this would mean they would have to jump around the arcane configurations set up to fix the laziness of packagers just so they can help maintain the product.

I have to think on this, because right here, right now, it's something I really, really, _don't_ want to support.

Reply:
I understand your point too, I think mainly about the end users who will end up with an "incomplete" build because of this "laziness".

The best "compromise" is maybe to put on the build guide all the available options with what they do provide
> BUILD_BROWSER provide browser docks and sources and also service integration if TWITCH_CLIENTID and/or RESTREAM_CLIENTID is set

And maybe supply "proper builds" with a solution like Open Build Service which can provide rpm, deb and archlinux/manjaro packages but it puts more work on you (obsproject members)

Reply:
We could implicitly enable all these features on Linux (always assuming ´BUILD_FOR_DISTRIBUTION` is set, but with `RelWithDebInfo` config), that would have side-effects for Linux maintainers though as visibility of cache variables might be affected (might also require additional code as I _guess_ people want to explicitly disable implicitly enabled features, which would bring back the mess I wanted to get away from).

Other possible choice would be to enable everything by default on every OS, but then I'd make more sense to rename all those `ENABLE` flags to `DISABLE` as that becomes the explicit choice.

I'd prefer the current state, but I'm fine with the team deciding against it.

Reply:
I think also we need to hear the opinion of people I mentioned earlier

Reply:
I'm the biggest point, after the packagers one, is maybe the real lack of options explanation in this [page](https://obsproject.com/wiki/Install-Instructions)

Reply:
> This is where a feature like pipewire (which I couldn't get to work in my tests) could be enabled by default. But afaik nobody took care of adopting the CI build scripts for pipewire support either, so IMO that PR should not have been merged in the first place and which is why it's not included here.

You are probably building an on old (read LTS) version of ubuntu, which is indeed incompatible with the pipewire plugin as written since it depends on a more or less bleeding edge version. So it was disabled in the ubuntu CI and only left enabled in the flatpak CI. And attempted to be explained in the build wiki https://obsproject.com/wiki/install-instructions#debian-based-build-directions though it should have been updated to disabling pipewire for older platforms now.

> (from below) Sane default for build trees is to deactivate as many optional features as possible

Im not sure I agree, but for the sake of argument, why are the things that you chose part of the minimal build? We could build precisely libobs and thats it unless you decide to enable other features like the UI or graphics backends.  Or to compare it to other platforms, why all wayland (Read: UI and Window capture) support default off? Why is webcam support default off? Arguably why is some audio capture default off? All of these are mandatory on other platforms so lets make it the same on linux. I personally think we should be consistent (functionality wise) across platforms in what we enable by default. Since we are rewriting it would be nice just do this, but at least we shouldnt step back from the small changes we have already merged towards this effort (without good reason).

I'm ok with deciding to switch to a single flag to enable all the (actually) optional features and asking all downstream package maintainers to use it (This is how some other build tools handle it for example). Perhaps there is already a default flag for this we can leverage.

Reply:
> I personally think we should be consistent (functionality wise) across platforms in what we enable by default

I'm fine with that, it just means that the idea of a "almost nothing active by default" build tree needs to be abandoned, but that's not a hill I need to die on.

So I'm happy to apply a list of plugins that should be active by default, I just need that list (for macOS I can do this myself, for Windows I can ask Ryan or Matt, for Linux I'm happy to take your advice, I'd need the same for FreeBSD then).

As for Pipewire, I'd like to know if we can get it to work on `ubuntu-latest` because I'd like the CI versions to very much be the "default" release variants.

Reply:
`libpipewire-0.3-dev` is available on 20.10, im not sure that the CI supports non-LTS so ubuntu-latest is probably actually not the latest.

Reply:
`ubuntu-latest` is Ubuntu 20.04 according to Github's docs. And I'd really advocate for not supporting a default build combination that cannot be used on CI, because that defeats the purpose of using CI in the first place.

Reply:
The issue with 20.10 is that xdg-portal-desktop does not require PipeWire, so PipeWire Screen Capture may not be available

Reply:
> And I'd really advocate for not supporting a default build combination that cannot be used on CI

I dont think anyone is advocating this. We have two CI tracks for linux now, and it's enabled in one of them so this is enough in my opinion.

Reply:
Linuuuuuxxxx!!!! \*shakes fist\*

I want to see you happy, so how about enabling PipeWire by default but only if Ubuntu 20.10+ is available - I'd just need the variant of `swvers` on Linux to detect that.

Reply:
im not sure what `swvers` and google doesnt appear very helpful unless you are talking about some node package. But I still think that it should be disabled explicitly if you dont have it, since this will end up being lost by package managers for months until users finally report it to them once their platform supports it.

If we wanted to make it optional we could just make the find_package optional.

Reply:
It is optional already, but that doesn't make sense then. I'll just enable it by default, make the packages required if it's enabled and disable it explicitly on CI.

Sorry for taking this long to come around, but I needed all that input to wrap my head around this issue.